### PR TITLE
Put semicolon at the end of the method declaration instead of "{…

### LINKED ISSUE #231: Do not show brackets for methods and constructos
Currently constructors and methods display usless { } at teh end of the line, e.g.:

```c#
public class IdentityClientOptions : ClientOptions {                                  
        public IdentityClientOptions() {}
}
```

It would be less busy to display it just like C#'s extern decls:

```c#
public class IdentityClientOptions : ClientOptions {                                  
        public IdentityClientOptions();
}
```

### PR DESCRIPTION
**Pull request by AlexGhiondea**
…hat is now.

The original intention was to have code that compiles, however having readable code is more important for the tool.

Fixes #231 

/cc @pakrym @KrzysztofCwalina

### CODE REVIEW

**Comment by pakrym** [comment]
I'm surprised tests didn't explode in CI but there are some and they should be fixed.

**Comment by pakrym** [comment]
image


This is why...

File: src/dotnet/APIView/APIView/Languages/CodeFileBuilder.cs [hunk 1]
```diff
@@ -319,9 +319,7 @@ private void BuildMember(CodeFileTokensBuilder builder, ISymbol member)
                 !member.IsAbstract &&
                 member.ContainingType.TypeKind != TypeKind.Interface)
             {
-                builder.Space();
-                builder.Punctuation(SyntaxKind.OpenBraceToken);
-                builder.Punctuation(SyntaxKind.CloseBraceToken);
+                builder.Punctuation(SyntaxKind.SemicolonToken);
```
Comment (pakrym) [inline src/dotnet/APIView/APIView/Languages/CodeFileBuilder.cs]: This method can be simplified, there are multiple branches printing semicolon and you don't need the complicated logic anymore.

Comment (AlexGhiondea) [reply src/dotnet/APIView/APIView/Languages/CodeFileBuilder.cs]: There are 2 branches that are printing the semicolon:
 - if you have a non-abstract method that is defined on a class
 - if you don't have a property

There are ways to optimize that code, but I am not sure if we want to do that. Unless we make the method print `;` in more situations, I don't know of a much cleaner way to write this.

Combining the statements looks weirder to me:
```csharp
if ((member.Kind == SymbolKind.Method && !member.IsAbstract && member.ContainingType.TypeKind != TypeKind.Interface) 
    || (member.Kind != SymbolKind.Property))
{
    builder.Punctuation(SyntaxKind.SemicolonToken);
}
```

**Comment by AlexGhiondea** [comment]
Hmm... This is becoming interesting :). The way the tests are written, there are a couple of issues:
- The code is expected to compile. By replacing the `{ }` with `;` the code we have in our tests will no longer compile
- Some of the tests validate that we correctly remove the content of the method -- that is probably no longer necessary.

I'm looking into this some more :)

File: src/dotnet/APIView/APIView/Languages/CodeFileBuilder.cs (hunk 1 above)
Comment (pakrym) [reply src/dotnet/APIView/APIView/Languages/CodeFileBuilder.cs]: I think you can entirely remove this branch and get the desired behavior.

**Pull request merged**